zio-schema/shared/src/main/scala/zio/schema/codec/ProtobufCodec.scala

```diff
@@ -587,6 +590,7 @@ object ProtobufCodec extends Codec {
         case StandardType.DoubleType => doubleDecoder
         case StandardType.BinaryType => binaryDecoder
         case StandardType.CharType   => stringDecoder.map(_.charAt(0))
+        case StandardType.UUIDType   => stringDecoder.map(UUID.fromString)
```

Comment:
Here I think we need to handle an invalid UUID string and return a `Left("invalid UUID")` So something like 

```scala
stringDecoder.flatMap { uuidString =>
  try {
   succeed(UUID.fromString(uuidString))
  } catch { case _: Throwable => fail("invalid UUID") }
}
```

Reply:
Ok will do, wasn't sure because I noticed that there wasn't any error handling on the `Date` types so I wasn't sure if that was supposed to be handled higher up.

Reply:
Updated! I had to add a new `succeedNow` to avoid the by-name parameter passing which would bypass the try catch.

Reply:
Looks great! I think you just need to run `sbt prepare` to make sure everything is formatted and we should be good to go.

Reply:
Done!